E0301 needs to be updated to new format

From: src/test/compile-fail/E0301.rs

E0301 needs a span_label, updating it from:

```
error[E0301]: cannot mutably borrow in a pattern guard
  --> src/test/compile-fail/E0301.rs:14:19
   |
14 |         option if option.take().is_none() => {}, //~ ERROR E0301
   |                   ^^^^^^
```

To:

```
error[E0301]: cannot mutably borrow in a pattern guard
  --> src/test/compile-fail/E0301.rs:14:19
   |
14 |         option if option.take().is_none() => {}, //~ ERROR E0301
   |                   ^^^^^^ borrowed mutably in pattern guard
```

--- a/src/librustc_const_eval/check_match.rs
+++ b/src/librustc_const_eval/check_match.rs
@@ -1175,8 +1175,10 @@ impl<'a, 'gcx, 'tcx> Delegate<'tcx> for MutationChecker<'a, 'gcx> {
               _: LoanCause) {
         match kind {
             MutBorrow => {
-                span_err!(self.cx.tcx.sess, span, E0301,
+                struct_span_err!(self.cx.tcx.sess, span, E0301,
                           "cannot mutably borrow in a pattern guard")
+                    .span_label(span, &format!("borrowed mutably in pattern guard"))
+                    .emit();
             }
             ImmBorrow | UniqueImmBorrow => {}
         }

--- a/src/test/compile-fail/E0301.rs
+++ b/src/test/compile-fail/E0301.rs
@@ -12,6 +12,7 @@ fn main() {
     match Some(()) {
         None => { },
         option if option.take().is_none() => {}, //~ ERROR E0301
+        //~| NOTE borrowed mutably in pattern guard
         Some(_) => { }
     }
 }

